### DISCUSSION TITLE
docs: Auto increment that is not a PrimaryKey

### DISCUSSION THREAD

**Discussion by pz129** [mikro-orm/mikro-orm#3502] [Q&A]
There should be a way to add an auto increment column that is not a `@PrimaryKey` if there is an `@Index` decorator.

If this is not possible, this may also be a bug.

**Comment by B4nan** [comment]
You can do this in postgres:

```ts
@Property({ autoincrement: true })
foo!: number;
```

https://github.com/mikro-orm/mikro-orm/blob/master/tests/features/schema-generator/serial-property.postgres.test.ts#L21-L22

Or what driver are you using?

**Reply by pz129** [reply]
I'm using MySQL, which I'm assuming it is not supported for then?

**Reply by B4nan** [reply]
In other drivers like MySQL you can have only a single autoincrement key in a table, right? So why would you not use that as the PK, what is your use case?

**Reply by pz129** [reply]
it's being used as a pagination key column which doesn't exist in all tables

while we use a UUID column as the PK across all tables

**Reply by B4nan** [reply]
You could go the other way, define your UUID column as a regular one with unique constraint, and have the autoincrement as primary key. I know it kinda sucks, but it should have the same effect :]

**Reply by pz129** [reply]
Is this something that could be supported? I think that it makes sense to have across other use cases.

One reason that this is not easy is that these tables extend off an Entity that defines a PrimaryKey already, and it would be very annoying to do so differently for these select few.

**Reply by B4nan** [reply]
I am affraid this is not supported at knex level, this would require the `create table` statement to include the `key` definition for such column, I dont think knex allows that, it creates indexes via separate queries.
